docs/local-development-environment.adoc

```diff
@@ -0,0 +1,182 @@
+version. Eg. once for SLES15-SP6 and again when trying with SLES15-SP7.
+====
+
+. Download a desired SLES KVM/Xen image from `download.suse.com` or
+from https://scc.suse.com[SUSE Customer Center] (you'd need to have a
+customer account for the later). Again, there is no SLES4SAP VM images
+already prepared from SUSE. The good thing, though, is that both
+systems share the same base installation. Example name for the image
+could be `SLES15-SP6-Minimal-VM.x86_64-kvm-and-xen-GM.qcow2`.
+
+. Once you have the desired SLES image give its path to the provided
+`make_box.sh` shell script in `devbox` directory:
++
+[source,bash]
+----
+$ ./devbox/make_sles_box.sh <path-to-sles-image.qcow2>
+----
+
+
+WARNING: Ignore permission warnings. They are caused by when `fuser`
+syscall is not running as root. In most cases, these warnings are
+harmless.
+
+. After the script finishes, a new `*.box`
+file is created in `devbox` directory with a name extracted from the
+name of the original image. You have to add that box file to
+Vagrant. For instance, if the name of the produced file is
+`SLES15-SP6.box`, then:
++
+[source,bash]
+----
+$ vagrant box add --name SLES15-SP6 ./devbox/SLES15-SP6.box
+----
+
+Now, Vagrant recognizes `SLES15-SP6` as valid base box.
+
+=== SLES4SAP virtual appliance
+
+At this point you can start Vagrant using "base" SLES but the VM would
+require _interactive_ initial setup and also will not be registered
+with SCC. Most importantly, it will not be SLES4SAP but rather base
+SLES.
+
+To handle all of that, a Combustion script would be used on first boot
+to "specialize" the base-box into a virtual appliance ready to be used
+as a test target for the Ansible playbook.
+
+Because the real Combustion file should contain some secrets, the
+`devbox` directory contains a script called
+`make_combustion_script.sh` that would generate the real startup
+script with the secrets baked in:
+
+[NOTE]
+====
+The following procedure should be executed only once if secrets remain
+the same, _regardless_ of the used SLES version image:
+====
+
+. Copy `devbox/secrets.tpl` as `devbox/secrets`.
+. Fill in  all the fields in `devbox/secrets`.
+. Generate the Combustion file by running the provided script:
++
+[source,bash]
+----
+$ ./devbox/make_combustion_script.sh
+----
+
+With this, all preparation steps are completed and a new VM could be
+started.
+
+== Usage with Vagrant
+
+Start the vagrant box with:
+
+[source,bash]
+----
+$ vagrant up
+----
+
+This will spawn a vagrant box with `SLES15-SP6` as base box by
+default. The base box can be changed by exporting
+`TRENTO_VAGRANT_BASE_BOX` env-var, for example:
+
+[source,bash]
+----
+$ export TRENTO_VAGRANT_BASE_BOX="SLES15-SP7"
+$ vagrant up
+----
+
+The provisioning will be automatic after the box starts. To force
+provision the vagrant box:
+
+[source,bash]
+----
+$ vagrant provision
+----
+
+Use this command when you want to re-provision the Vagrant box (re-run
```

Comment:
What about a more active voice: 
Use this command to re-provision the Vagrant box (this will re-run the Ansible playbook). You might need it, for example, when changing variables during the development process.